test/common_utils.py

```diff
@@ -565,14 +552,68 @@ def remove_prefix(text, prefix):
         munged_id = remove_prefix(self.id(), module_id + ".")
         test_file = os.path.realpath(sys.modules[module_id].__file__)
         expected_file = os.path.join(os.path.dirname(test_file),
-                                     "expect",
+                                     "example",
                                      munged_id)
 
         subname_output = ""
         if subname:
             expected_file += "-" + subname
             subname_output = " ({})".format(subname)
-        expected_file += ".expect"
+        expected_file += ".example"
+        return expected_file, munged_id, subname_output
+
+    def updateExampleExpect(self, s, subname=None):
+        r"""
+        Updates the file derived from the name of this test and subname to
+        the string. This does not test against the existing output and exists for
+        documentation and to signal unintended changes. This file is placed in
+        the 'example' directory in the same directory as the test script.
+
+        If you call this multiple times in a single function, you must
+        give a unique subname each time.
+        """
+        if not (isinstance(s, str) or (sys.version_info[0] == 2 and isinstance(s, unicode))):
+            raise TypeError("updateExampleExpect is strings only")
+
+        expected_file, munged_id, subname_output = self.getFileIdSubname("example", subname)
+        expected = None
+
+        def updateOutput(start_string):
+            print("{} for {}{}:\n\n{}".format(start_string, munged_id, subname_output, s))
+            with open(expected_file, 'w') as f:
+                f.write(s)
+
+        try:
+            with open(expected_file) as f:
+                expected = f.read()
+        except IOError as e:
+            if e.errno != errno.ENOENT:
+                raise
+            return updateOutput("Setting example output")
+
+        # a hack for JIT tests
```

Comment:
can you comment a little bit more on why we need this hack? good for future reference.

Reply:
This was existing code that I refactored... i'll look into why we need it